src/main/java/seedu/address/model/person/Date.java

```diff
@@ -32,7 +34,7 @@ public abstract class Date {
      */
     public Date(String date) {
         checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
-        value = date;
+        value = parseLocalDate(date).format(DATE_TIME_FORMATTER);
```

Comment:
Since the `date` input string is now case-insenstive, this additional parsing ensures that the month in `value` instance field is capitalised, e.g. `20 jan 2019` becomes `20 Jan 2019`, `31 DEC 1980` becomes `31 Dec 1980`.